src/datasets/features/audio.py

```diff
@@ -95,6 +95,11 @@ def decode_example(self, value: dict) -> dict:
             raise ValueError(f"An audio sample should have one of 'path' or 'bytes' but both are None in {value}.")
         elif path is not None and path.endswith("mp3"):
             array, sampling_rate = self._decode_mp3(file if file else path)
+        elif path is not None and path.endswith("opus"):
```

Comment:
I can check if a file is in opus format only if `path` is not None. actually I don't understand in what we can have an empty path (I though that in the loading script we always provide either full path to file or relative path)? But I saw that audio feature tests check the case with empty path.

Reply:
It can happen when the file is not available locally (e.g. in streaming mode) - in this case one can pass a path or just the file name so that we can know the extension of the file, and therefore be able to decode it.

Note that in the future we can stop using `path` to check whether a file is in OPUS or MP3 or whatever, and check the [magic number](https://en.wikipedia.org/wiki/List_of_file_signatures) of the file instead